server/s3_motr_writer.h

```diff
@@ -119,6 +119,11 @@ class S3MotrWiter {
   // buffer currently used to write, will be freed on completion
   std::shared_ptr<S3AsyncBufferOptContainer> write_async_buffer;
 
+  // used for checksum calculation testing
+  // fill entire object with zeroes after checksum calculation, but before
+  // writing to Motr
+  bool corrupt_fill_zero = false;
```

Comment:
Presumably there is an easy way to set this without having to change and recompile the code, right?

Reply:
Right. Just enable `S3_ST_MD5_ERROR_INJECT_ENABLED` and pass an object with 'z' in the first byte of the object.